bindings/c/fdb_c.cpp

```diff
@@ -901,6 +911,41 @@ FDBFuture* fdb_transaction_get_range_impl(FDBTransaction* tr,
 	                    .extractPtr());
 }
 
+extern "C" DLLEXPORT FDBFuture* fdb_transaction_get_mapped_range_v2(FDBTransaction* tr,
+                                                                    uint8_t const* begin_key_name,
+                                                                    int begin_key_name_length,
+                                                                    fdb_bool_t begin_or_equal,
+                                                                    int begin_offset,
+                                                                    uint8_t const* end_key_name,
+                                                                    int end_key_name_length,
+                                                                    fdb_bool_t end_or_equal,
+                                                                    int end_offset,
+                                                                    uint8_t const* mapper_name,
+                                                                    int mapper_name_length,
+                                                                    int limit,
+                                                                    int target_bytes,
+                                                                    FDBStreamingMode mode,
+                                                                    int iteration,
+                                                                    fdb_bool_t snapshot,
+                                                                    fdb_bool_t reverse,
+                                                                    uint8_t const* mapped_range_params,
```

Comment:
the verification is done at the server side. and the format is documented as well in  https://github.com/apple/foundationdb/pull/9257/commits/8d2665771a4eb1f3341947765ea1959418cf03ed#diff-7277186ef3b9f2a1e737152de069c3aff3e2d92e5c99e5caccc9330471e79b63R18-R26, in `fdbclient/include/fdbclient/MappedRange.h`.

Reply:
reply is [here](https://github.com/apple/foundationdb/commit/8d2665771a4eb1f3341947765ea1959418cf03ed#diff-7277186ef3b9f2a1e737152de069c3aff3e2d92e5c99e5caccc9330471e79b63R36-R45)

Reply:
I see. It might be good to allude to that code somewhere else so that it's more clear what the contract is supposed to be. Also, I'm still not sure I understand the format, even having read that documentation. In particular, it's unclear to me what the "version" refers to, and what the "type codes" are supposed to represent. If we have something that can parse that information and put it into something that can be more easily understood, then maybe that's fine.

I will reiterate, though, that we have to be a little careful with an API like this, because it is effectively using a byte array as an end-run around the usual API versioning scheme. So, for example, if in 7.4, it's decided that we want to have this return a different set of values with a different "version", then we need to make sure that if the user specifies an older API version that we still return data that looks the old way (I think).

Reply:
yes, we are on the same page here, the server has to recognize the version first, then parse the version accordingly, so server will have something like
```
v = parseVersion(bytes);
if (v == 2) {
   parseV2()
} else if (v == 3) {
   parseV3()
} else {
 // throw unsupported_version();
}
```


I can certainly make the document better, let me just resolve the main issue, that is whether to use the existing mechanism for API versioning first, please check my comment above.